.travis.yml

```
@@ -7,22 +8,72 @@ python:
   - "3.4"
   - "3.5"
   - "3.6"
-  - "pypy"
+  - "3.7"
+  - "pypy2.7-6.0"
+  - "pypy3.5-6.0"
 
-matrix:
+env:
+  global:
+    LIBGIT2: ~/libgit2/_install/
+    LD_LIBRARY_PATH: ${LIBGIT2}/lib:${LD_LIBRARY_PATH}
+    LIBGIT2_VERSION: "0.27"
+
+
+jobs:
   include:
-    - python: 3.7
-      dist: xenial
-      sudo: true
+  - stage: &deploy_stage Publish to PyPI (runs only for tagged commits)
+    name: >-
+      OS-specific manylinux1 wheels
+    python: "3.7"
+    services:
+    - docker
+    install: skip
+    script: &docker-script
+    - travis/build-all-manylinux1-wheels.sh "pygit2" "${LIBGIT2_VERSION}"
+    before_deploy:
+    - ls dist
+    - pip install twine
+    - twine check dist/*
+    deploy:
+      provider: pypi
+      # `skip-cleanup: true` is required to preserve binary wheels, built
+      # inside of manylinux1 docker container during `script` step above.
+      skip-cleanup: true
+      # `skip-existing: true` is required to skip uploading dists, already
+      # present in PYPI instead of failing the whole process.
+      # This happens when other CI (AppVeyor etc.) has already uploaded
+      # the very same dist (usually sdist).
+      skip-existing: true
+      user: jdavid
+      password:
+        # Encrypted with `travis encrypt -r aio-libs/aiohttp --api-endpoint 'https://api.travis-ci.com/'`:
+        secure: >-
+          TODO: REPLACE THIS WITH AN ENCRYPTED PASSWORD
+      # Although Travis CI instructs `setup.py` to build source distribution,
+      # which is default value for distribution option (`distribution: sdist`),
+      # it will also upload all wheels we've previously built in manylinux1
+      # docker container using `twine upload -r pypi dist/*` command.
+      # Also since commit https://github.com/travis-ci/dpl/commit/90b5e39
+      # it is default that Travis PYPI provider has `skip_upload_docs: true`
+      # set by default.
+      # Besides above, we don't do cleanup of `dist/*`, because it's being done
+      # by Travis CI PYPI deployment provider after upload, unconditionally.
+      on:
+        tags: true
+        all_branches: true
 
-env: LIBGIT2=~/libgit2/_install/ LD_LIBRARY_PATH=~/libgit2/_install/lib
 
 before_install:
   - sudo apt-get install cmake
-  - "./.travis.sh"
+  - ./.travis.sh "${LIBGIT2_VERSION}"
 
 install:
   - pip install .
 
 script:
   - pytest
+
+stages:
+- test
+- name: *deploy_stage
+  if: 1=1 OR tag IS present OR type IN (cron, api)
```

@jdavid
We'll need to apply
```suggestion
  if: tag IS present OR type IN (cron, api)
```
once password is updated.